presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PullUpExpressionInLambdaRules.java

```diff
@@ -213,6 +213,7 @@ private static class ValidExpressionExtractor
         // Bind expression will complicate the lambda expression, we apply this optimization before DesugarLambdaRule. And if there are bind expression, skip
         private static final List<SpecialFormExpression.Form> UNSUPPORTED_TYPES = ImmutableList.of(SpecialFormExpression.Form.BIND);
         private static final List<Class<?>> SUPPORTED_JAVA_TYPES = ImmutableList.of(boolean.class, long.class, double.class, Slice.class, Block.class);
+        private static final int MAX_NUM_OF_ARGUMENTS = 1_000;
```

Comment:
is that a realistic upper limit? should we have something smaller?

Reply:
Removed the limit, I think it's not needed now.